warehouse/forklift/legacy.py

```diff
@@ -402,6 +402,35 @@ def _validate_classifiers(form, field):
             )
 
 
+def _validate_dynamic(_form, field):
+    declared_dynamic_fields = set(map(str.title, field.data or []))
+    disallowed_dynamic_fields = {"Name", "Version", "Metadata-Version"}
```

Comment:
From https://github.com/pypi/warehouse/pull/11380#discussion_r1033894111:

PEP 643 only says:

> The fields `Name` and `Version` MUST NOT be marked as Dynamic.

Should `Metadata-Version` be included here? Or should the PEP be updated?

Reply:
This is already the case in [the spec](https://packaging.python.org/en/latest/specifications/core-metadata/#dynamic-multiple-use). It was added in https://github.com/pypa/packaging.python.org/pull/1057 (I can't check this right now as github is playing up, but that's what git blame tells me locally!)

Reply:
The principle is that the spec is the definitive document, and the PEP is historical, so there's no need to update the PEP.

Reply:
I'll defer to that!